include/core/evaporation_model.h

```diff
@@ -0,0 +1,707 @@
+   * be calculated
+   * @param jacobian Vector of the value of the derivative of the recoil pressure
+   * with respect to the field id
+   */
+  void
+  vector_momentum_flux_jacobian(
+    const std::map<field, std::vector<double>> & /*field_vectors*/,
+    const field /*id*/,
+    std::vector<double> &jacobian_vector) override
+  {
+    std::fill(jacobian_vector.begin(), jacobian_vector.end(), 0);
+  }
+
+private:
+  const double n_evaporation;
+  const double latent_heat_evaporation;
+  const double ambient_gas_density_inv;
+  const double liquid_density_inv;
+};
+
+/**
+ * @brief EvaporationModelTemperature. Class that allows to calculate the
+ * evaporation heat and momentum fluxes when a temperature dependent mass flux
+ * is considered.
+ *
+ * @param p_evaporation evaporation model parameters.
+ */
+class EvaporationModelTemperature : public EvaporationModel
+{
+public:
+  EvaporationModelTemperature(const Parameters::Evaporation &p_evaporation)
+    : evaporation_coefficient(p_evaporation.evaporation_coefficient)
+    , recoil_pressure_coefficient(p_evaporation.recoil_pressure_coefficient)
+    , molar_mass(p_evaporation.molar_mass)
+    , boiling_temperature_inv(1.0 / p_evaporation.boiling_temperature)
+    , latent_heat_evaporation(p_evaporation.latent_heat_evaporation)
+    , ambient_pressure(p_evaporation.ambient_pressure)
+    , ambient_gas_density_inv(1.0 / p_evaporation.ambient_gas_density)
+    , liquid_density_inv(1.0 / p_evaporation.liquid_density)
+  {
+    model_depends_on[temperature] = true;
+    this->R_inv                   = 1.0 / 8.3145;
+    this->L_vap_x_M_x_R_inv =
+      latent_heat_evaporation * molar_mass * this->R_inv;
+    this->M_x_2PI_inv_x_R_inv = molar_mass / (2.0 * M_PI) * this->R_inv;
+  }
+
+  /**
+   * @brief saturation_pressure Calculates the value of the staturation pressure.
+   * @param fields_value Value of the various field on which the flux may depend.
+   * @return value of the flux calculated with the fields_value
+   */
+  double
+  saturation_pressure(const std::map<field, double> &fields_value)
+  {
+    const double temperature_inv =
+      1.0 / (fields_value.at(field::temperature) + 1e-16);
+
+    return ambient_pressure *
+           std::exp(-L_vap_x_M_x_R_inv *
+                    (temperature_inv - boiling_temperature_inv));
+  }
+
+  /**
+   * @brief vector_mass_flux Calculates the values of the staturation pressure
+   * for multiple points
+   * @param field_vectors Value of the various fields on which the flux may depend.
+   * @param mass_flux_vector Vectors of the mass flux values
+   */
+  void
+  vector_saturation_pressure(
+    const std::map<field, std::vector<double>> &field_vectors,
+    std::vector<double>                        &saturation_pressure_vector)
+  {
+    const std::vector<double> &temperature =
+      field_vectors.at(field::temperature);
+
+    for (unsigned int i = 0; i < saturation_pressure_vector.size(); ++i)
```

Comment:
It may be faster to declare an `unsigned int vector_size` first instead of calling size() many times, but whether its significant I don't know

Reply:
I have the same comment for the other functions, but I won't rewrite it